src/js/components/DataFilters/DataFilters.js

```diff
@@ -112,44 +117,59 @@ export const DataFilters = ({
     }
   }
 
+  const contextValue = useMemo(() => ({ pendingReset }), []);
```

Comment:
So the context itself never changes and won't cause a re-render since it's a ref that itself never changes. So it's relying on the `touched` to cause any updates on the badge and in theory the pending reset flag on that field in the Set would have been updated before that. Did I get that right?

Reply:
Yes, that is correct. Since changing the range selector value will trigger a "touched" change, that's how the process is working.